docs/basic-features/environment-variables.md

```diff
@@ -109,15 +109,11 @@ Next.js allows you to set defaults in `.env` (all environments), `.env.developme
 
 ## Environment Variables on Vercel
 
-When deploying on [Vercel](https://vercel.com) you can configure secrets in the [Environment Variables](https://vercel.com/docs/v2/build-step#environment-variables) section of the project in the Vercel dashboard.
+When deploying your Next.js application to [Vercel](https://vercel.com), Environment Variables can be configured [in the Project Settings](https://vercel.com/docs/environment-variables).
 
-You can still use `.env`, `.env.development` and `.env.production` to add defaults.
+All kinds of Environment Variables should be configured there. Even Environment Variables used in Development – which can afterwards be [downloaded onto your local device](https://vercel.com/docs/environment-variables#development-environment-variables).
 
-If you've configured [Development Environment Variables](https://vercel.com/docs/v2/build-step#development-environment-variables) you can pull them into a `.env.local` for usage on your local machine using the following command:
-
-```bash
-vercel env pull .env.local
-```
+On Vercel, `.env` should only be used [in Development](https://vercel.com/docs/environment-variables#development-environment-variables). **Do not deploy `.env` or any other variation of it.**
```

Comment:
The problem with using `.env`, if i remember correctly, is that it will be deployed when you run `vc` (unless you add it to `.vercelignore`).

Reply:
It'll be deployed when you have it in the git repo as well, so the statement at the end is incorrect. https://github.com/vercel/next.js/pull/20708#issuecomment-753627263